webview-ui/src/Draft/DraftWorkflow/dataLoading.ts

```diff
@@ -0,0 +1,220 @@
+import {
+    AcrKey,
+    ClusterKey,
+    GitHubRepo,
+    GitHubRepoKey,
+    Subscription,
+    acrKeysMatch,
+    clusterKeysMatch,
+} from "../../../../src/webview-contract/webviewDefinitions/draft/types";
+import { getOrThrow } from "../../utilities/array";
+import { Lazy, isLoaded, isNotLoaded, map as lazyMap, newNotLoaded } from "../../utilities/lazy";
+import { EventHandlers } from "../../utilities/state";
+import {
+    AcrReferenceData,
+    AzureReferenceData,
+    ClusterReferenceData,
+    GitHubRepositoryReferenceData,
+    GitHubReferenceData,
+    SubscriptionReferenceData,
+} from "../state/stateTypes";
+import { EventDef, vscode } from "./state";
+
+export type EventHandlerFunc = (eventHandlers: EventHandlers<EventDef>) => void;
+
+export function ensureBranchNamesLoaded(
+    referenceData: GitHubReferenceData,
+    repo: GitHubRepo | null,
```

Comment:
I think you're suggesting starting the function with:
```typescript
if (repo === null) {
    return newNotLoaded();
}
...
```

Then `getGitHubRepoReferenceData` could take a `repo` parameter that was not nullable.

That's certainly what I'd normally prefer to do.

However, all these functions intentionally take nullable parameters, specifically to avoid `null`-checks at the calling site, which would significantly increase the amount of code.